Google.Api.Gax.Grpc.Tests/Rest/RpcCancellationContextTest.cs

```diff
@@ -0,0 +1,157 @@
+﻿/*
+ * Copyright 2022 Google LLC
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using Grpc.Core;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Api.Gax.Grpc.Rest.Tests;
+
+public class RpcCancellationContextTest
```

Comment:
Maybe add one or two tests that receive a null deadline cancellation token source, mostly for documentation and so that we don't inadvertenly change that behaviour later.

Reply:
Done.